code/modules/power/singularity/singularity.dm

```diff
@@ -371,7 +371,7 @@
 			return 0
 	else if(locate(/obj/machinery/shieldwallgen) in T)
 		var/obj/machinery/shieldwallgen/S = locate(/obj/machinery/shieldwallgen) in T
-		if(S && S.active)
+		if(S && S.activated) // Check this
```

Comment:
```suggestion
		if(S && S.activated)
```
It's fine. I missed line 366

Reply:
Cant you also do `if(S?.activated)` here too?

Reply:
Yep totally can here. I'll change it once I can